src/visitors/semantic/lint.rs

```diff
@@ -191,3 +191,44 @@ impl<'a> flux::semantic::walk::Visitor<'a>
         true
     }
 }
+
+pub struct InfluxDBIdentifierDiagnosticVisitor {
+    names: Vec<String>,
+    pub diagnostics: Vec<(Option<String>, lsp::Diagnostic)>,
+}
+
+impl Default for InfluxDBIdentifierDiagnosticVisitor {
+    fn default() -> Self {
+        Self {
+            diagnostics: vec![],
+            names: vec!["v".into(), "task".into(), "params".into()],
+        }
+    }
+}
+
+impl<'a> flux::semantic::walk::Visitor<'a>
+    for InfluxDBIdentifierDiagnosticVisitor
+{
+    fn visit(&mut self, node: WalkNode<'a>) -> bool {
+        if let WalkNode::VariableAssgn(assign) = node {
+            if self.names.contains(&assign.id.name.to_string()) {
```

Comment:
Question. Is it also possible to MemberAssignment which tries to assign a given `v.<prop>`? Do we also need to handle this use case?

Reply:
I thought about that, and don't think we need to handle that case. We're trying to prevent people from tripping over the mystery meat functionality, not hook into it and do things (at that point, no amount of linting will ever help).